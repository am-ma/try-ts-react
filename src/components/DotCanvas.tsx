import React from 'react';
import Dot from './Dot';
import DotValues from '~/models/DotValues.model';

interface DotCanvasProps {
  onClick: Function;
  dots: DotValues[];
  height: number;
  width: number;
}

interface DotCanvasState {}

export default class DotCanvas extends React.Component<DotCanvasProps, DotCanvasState> {
  renderDot(index: number) {
    const values = this.props.dots[index];
    return <Dot values={values} key={index} onClick={() => this.props.onClick(index)} />;
  }

  render() {
    const dots = Array(this.props.height)
      .fill(null)
      .map((_, h) => (
        <div className="canvas-row" key={h}>
          {Array(this.props.width)
            .fill(null)
            .map((_, w) => {
              const index = w + h * this.props.width;
              return this.renderDot(index);
            })}
        </div>
      ));

    return <div>{dots}</div>;
  }
}
