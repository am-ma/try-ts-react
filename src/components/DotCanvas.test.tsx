import React from 'react';
import DotCanvas from './DotCanvas';
import DotValues from '~/models/DotValues.model';
import renderer from 'react-test-renderer';

describe('DotCanvas', () => {
  describe('render', () => {
    it('will render', () => {
      const dot1: DotValues = { isFill: true, color: '#000' };
      const dot2: DotValues = { isFill: true, color: '#000' };
      const dot3: DotValues = { isFill: false, color: '#000' };
      const dot4: DotValues = { isFill: false, color: '#000' };
      const dots = [dot1, dot2, dot3, dot4];
      const onClick = jest.fn();

      const result = renderer.create(<DotCanvas onClick={onClick} dots={dots} height={2} width={2} />);
      expect(result).toMatchSnapshot();
    });
  });
});
