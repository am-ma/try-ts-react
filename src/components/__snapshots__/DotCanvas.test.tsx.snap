// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotCanvas render will render 1`] = `
<div>
  <div
    className="canvas-row"
  >
    <button
      className="dot"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#000",
        }
      }
    />
    <button
      className="dot"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#000",
        }
      }
    />
  </div>
  <div
    className="canvas-row"
  >
    <button
      className="dot"
      onClick={[Function]}
      style={Object {}}
    />
    <button
      className="dot"
      onClick={[Function]}
      style={Object {}}
    />
  </div>
</div>
`;
